fix(ui): Add missing ownership rules for issue owner suggestions

taylangocmen opened the pull request:
This adds:
- tags to the ownership rules in issue owner suggestions,
- a collapsed list for commits and rules, both clipped after 3 with a view more button
- also changes the look of the ownership rules, makes it a code tag entirely.

Addressing: https://app.asana.com/0/1174100696963347/1174801027541739/f

DESIGN:
<img width="1324" alt="tag-owner-rules-design" src="https://user-images.githubusercontent.com/15015880/82718732-27aea200-9c59-11ea-9cbe-d12ff92946ee.png">

OLD:
![Screen Shot 2020-05-22 at 5 51 00 PM](https://user-images.githubusercontent.com/15015880/82718669-bb33a300-9c58-11ea-825d-08cf74d632b7.png)

NEW:
![May-26-2020 15-09-41](https://user-images.githubusercontent.com/15015880/82955407-a199a080-9f63-11ea-8461-56b5de348a55.gif)

adhiraj commented:
What if we change the header in the hovercard from "ownership rules" to "matching ownership rules" to make it clearer?

adhiraj commented:
Because it's in a "suggested assignees" section and without the "matching" it could be a bit of a mental jump

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -158,7 +185,16 @@ const CommitDate = styled(({date, ...props}) => (
-const ReasonItem = styled('div')`
+const CommitReasonItem = styled('div')`
+  display: flex;
+  align-items: flex-start;
```
Is this default?

taylangocmen replied:
Default is stretch, without this tags are stretched, when children of this are taller than 1 lh:
![Screen Shot 2020-05-22 at 3 42 11 PM](https://user-images.githubusercontent.com/15015880/82714335-f4f9af00-9c42-11ea-92ab-08f0dbc42133.png)

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -14,125 +14,152 @@ import SentryTypes from 'app/sentryTypes';
+    rules: PropTypes.arrayOf(PropTypes.array),
+  };
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      commitsExpanded: false,
+      rulesExpanded: false,
+    };
+  }
```
Instead of `constructor`, let's define/initialize `state` as a class property

taylangocmen replied:
Will do

billyvg replied:
Still needs this

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -14,125 +14,152 @@ import SentryTypes from 'app/sentryTypes';
+                    ))}
+                </div>
+                {rules.length > 3 && !this.state.rulesExpanded ? (
+                  <ViewMoreButton onClick={() => this.setState({rulesExpanded: true})} />
```
Personally, I prefer seeing this more explicitly laid out (unless it's re-used a few times):

```
<ViewMoreButton>
  <ViewMoreLink onClick={}>View More</ViewMoreLink>
</ViewMoreButton>
```

Otherwise we have to jump around a bit between the render and the styled component. SCs should mostly be just css.

taylangocmen replied:
Fixed with other changes

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +215,22 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
 `;
 
+const ViewMoreButton = styled(({onClick, ...props}) => (
+  <div {...props}>
+    <ViewMoreLink onClick={onClick} />
+  </div>
```
Should onClick be on the wrapper for a larger click area?

taylangocmen replied:
Will do and make it a button

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +215,22 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
+  width: 100%;
+  text-align: right;
+`;
+const ViewMoreLink = styled(p => <a {...p}>View more</a>)`
```
Can we use a `<Button>` with priority=zero here?

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +215,22 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
+  text-align: right;
+`;
+const ViewMoreLink = styled(p => <a {...p}>View more</a>)`
+  color: #9586a5;
```
Hmm should we be using a custom color?

taylangocmen replied:
It's the h6 color we are using for the titles, which was the case in design, it can be gray2, they are very close

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +215,22 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
+`;
+const ViewMoreLink = styled(p => <a {...p}>View more</a>)`
+  color: #9586a5;
+  font-size: 11px;
```
Let's use a font size from theme.

taylangocmen replied:
will do

taylangocmen commented:
> What if we change the header in the hovercard from "ownership rules" to "matching ownership rules" to make it clearer?

Changed the title as suggested, made the `Viev more` a button, changed to theme styles instead of custom styles.

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +215,21 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
 `;
 
+const ViewMoreButton = styled(p => (
+  <button className="btn btn-link" {...p}>
```
Can we use the `<Button>` component here? We're phasing out the use of `btn` class names (unless it's used in an old button group). `<Button>` is pretty opinionated for padding, but I think we may be able to use `priority=link` w/ `size="zero"` - but if it's a pain, then we can use `Link`.

taylangocmen replied:
Works well

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +215,21 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
 
+const ViewMoreButton = styled(p => (
+  <button className="btn btn-link" {...p}>
+    View more
```
`t()`

taylangocmen replied:
My mistake updated

taylangocmen commented:
Updated the PR and the screenshot to reflect the changes

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -7,132 +7,157 @@ import {t, tct} from 'app/locale';
+            {commits !== undefined && (
+              <React.Fragment>
+                <div className="divider">
+                  <h6>{t('Suspect Commits')}</h6>
```
Are we sure we want to change this? `Suspect Commits` is a slightly different feature

taylangocmen replied:
That was what Adi suggested, we can revert it

adhiraj replied:
hmm really? I see that I just mentioned changing "ownership rules" to "matching ownership rules"

taylangocmen replied:
I don't know why I read it as that, I will change the titles to `Commits` and `Matching Ownership Rules`

billyvg commented on src/sentry/static/sentry/app/components/group/suggestedOwnerHovercard.jsx:
```diff
@@ -179,6 +213,20 @@ const OwnershipTag = styled(({tagType, ...props}) => <div {...props}>{tagType}</
+  </Button>
+))`
+  border: none;
+  color: ${p => p.theme.gray2} !important;
```
Can we do this without `!important` now that we're using the Button component?

benvinegar commented:
Congrats!
